cmake/helpers/GdalGenerateConfig.cmake

```diff
@@ -21,24 +21,45 @@ function(gdal_join_and_quote _var)
     set(${_var} "${string}" PARENT_SCOPE)
 endfunction()
 
-# Return a flat list of libs including target linking requirements.
-function(gdal_flatten_link_libraries _result)
+# Return a flat list of libs including target linking requirements,
+# prepended to _result input. Targets appear only once, as late as possible.
+# Targets which are already in _result are not resolved/added again.
+function(gdal_flatten_link_libraries _result _is_debug_build)
+    set(_libs "${${_result}}")
+    gdal_process_link_keywords(working_list "${_is_debug_build}" ${ARGN})
+    while(working_list)
+        list(POP_BACK working_list _lib)
+        if(NOT TARGET "${_lib}")
+            list(PREPEND _libs "${_lib}")
+        elseif(_lib IN_LIST _libs)
+            # already resolved, do not repeat
+            continue()
```

Comment:
Before this PR, the target `_lib` would have been visited every time.
With the resolved tail of link `_libs`, finding a target in the tail means its set of link libs is already there, too.